docs/nydus-zran.md

```diff
@@ -0,0 +1,56 @@
+# Nydus zran artifact user guide
+
+This guide shows how to generate a smaller **nydus zran artifact** from an existing OCI v1 image, which is used to accelerate the image directly. It has the advantage of eliminating the need to store two images and generating much faster than converting the full accelerated image, which can save image registry storage space compared to the native nydus image. A simple test result as follows:
+
+- Image: node:19.0
+- Workload: node -v
+- Registry Network: 3MB/s
+
+| image type             | image size | acceld conversion | acceld push | nerdctl run | read data |
+| ---------------------- | ---------- | ----------------- | ----------- | ----------- | --------- |
+| OCI v1                 | 353.05MB   | -                 | -           | 126s        | 353.05MB  |
+| Nydus (Native RAFS v6) | 337.94MB   | 29s               | 1m58s       | 11s         | 21.18MB   |
+| Nydus (Zran)           | 14MB       | 11s               | 12s         | 15s         | 28.78MB   |
+
+## Generate nydus zran artifact
+
+1. Build `nydus-image` and `nydusd`:
+
+``` bash
+git clone https://github.com/dragonflyoss/image-service.git
+
+# Install nydusd
+cargo build --target x86_64-unknown-linux-musl --release --bin nydusd
+sudo cp ./target/x86_64-unknown-linux-musl/release/nydusd /usr/bin/nydusd
+
+# Install nydus-image
+cargo build --target x86_64-unknown-linux-musl --release --bin nydus-image
+sudo cp ./target/x86_64-unknown-linux-musl/release/nydus-image /usr/bin/nydus-image
+```
+
+2. Build `accelctl`:
+
+``` bash
+git clone https://github.com/imeoer/acceleration-service.git -b nydus-zran-ref
+cd acceleration-service
+# Got compiled ./accelctl
+make
+```
+
+3. Generate nydus zran artifact：
+
+Following the comments in `./misc/config/config.nydus.ref.yaml` configuration file, ensure base64 encoded registry auth is configured correctly.
+
+``` bash
+# Convert the existing OCI v1 image `your-registry.com/node:19.0` to `your-registry.com/node:19.0-nydus-ref`
+sudo ./accelctl convert --config ./misc/config/config.nydus.ref.yaml your-registry.com/node:19.0
+```
+
+## Run nydus zran artifact：
+
+Follow the [doc](https://github.com/dragonflyoss/image-service/blob/master/docs/containerd-env-setup.md) to configure `containerd` and `nydus-snapshotter` (containerd-nydus-grpc)：
+
+``` bash
+# Run nydus zran artifact
+sudo nerdctl --snapshotter nydus run --rm -it your-registry.com/node:19.0-nydus-ref node -v
+```
```

Comment:
Let's pre-convert some zran images and upload them to ghcr.io so that users can easily test it.

Reply:
Added